lib/ui_components/buttons/save_request_button.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:first_project/enums/leave_request_status.dart';
+import 'package:first_project/extensions/context_extensions/colors.dart';
+import 'package:first_project/extensions/context_extensions/text_styles.dart';
+import 'package:first_project/models/leave_request.dart';
+import 'package:first_project/providers/date_map_state_provider.dart';
+import 'package:first_project/providers/leave_type_state_provider.dart';
+import 'package:first_project/providers/reason_state_provider.dart';
+import 'package:first_project/providers/request_visibility_state_provider.dart';
+import 'package:first_project/providers/user_notifier_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:hive/hive.dart';
+
+class SaveRequestButton extends ConsumerWidget {
+  const SaveRequestButton({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    return SizedBox(
+      width: 96,
+      child: TextButton(
+        style: ButtonStyle(
+            backgroundColor: WidgetStatePropertyAll(context.secondary),
+            shape: WidgetStatePropertyAll(
+              RoundedRectangleBorder(borderRadius: BorderRadius.circular(8)),
+            )),
+        onPressed: () {
+          final requestBox = Hive.box<LeaveRequest>('requestBox');
+          final user = ref.read(userStateProvider);
+          final startDate = ref.read(dateMapStateProvider).values.first;
```

Comment:
It is valid to have multiple state providers like what you did here.

Another better option is to have a single notifier that holds the state of everything. In places where you need to listen to only one param change you can use `ref.watch(provider.select())`